Source/Meadow.Contracts/ExtensionMethods.cs

```diff
@@ -122,40 +137,35 @@ public static void Fire(this Delegate handler, params object[] args)
         }
     }
 
-    public static void Fire(this EventHandler h, object sender)
+    /// <summary>
+    /// Traverses the handler's invocation list and invokes each
+    /// </summary>
+    /// <param name="handler">The eventhandler we are acting upon.</param>
+    /// <param name="sender">The sender object.</param>
+    public static void Fire(this EventHandler handler, object sender)
     {
-        Fire(h, sender, EventArgs.Empty);
+        Fire(handler, sender, EventArgs.Empty);
     }
 
+    /// <summary>
+    /// Traverses the handler's invocation list and invokes each
+    /// </summary>
+    /// <param name="handler">The eventhandler we are acting upon.</param>
+    /// <param name="sender">The sender object.</param>
+    /// <param name="args">The arguments we want to pass to each delegate.</param>
     public static void Fire(this EventHandler handler, object sender, EventArgs args)
     {
-        if (handler == null) return;
-        foreach (var d in handler.GetInvocationList())
-        {
-            try
-            {
-                d.DynamicInvoke(sender, args);
-            }
-            catch (Exception ex)
-            {
-                Resolver.Log.Error($"Event handler threw {ex.GetType().Name}: {ex.Message}");
-            }
-        }
+        Fire(handler, sender, args);
```

Comment:
Does this not just call itself?

Reply:
@adrianstevens I was working on this at 2am, and obviously had a brain fart. You are absolutely correct.

Reply:
@adrianstevens I now have the centralised code I was trying to achieve. I don't think we'll have recursion with this one.